Docs/docs.js

```diff
@@ -0,0 +1,141 @@
+
+try{
+    //To toggle the main sidebar
+    const mainSidebar = document.querySelector("aside.tr-drawer-modal");
+    const hamburgers = document.querySelectorAll(".hamburger");
+    
+    hamburgers.forEach(btn=>btn.addEventListener('click', e => {
+        let toggleTarget = document.getElementById(btn.getAttribute('data-target'));
+        toggleTarget.classList.toggle('active');
+
+        if(toggleTarget.className.includes('tr-drawer-modal')){
+            toggleTarget.parentElement.classList.toggle('active') //To activate the BG overlay
+        }
+        e.stopPropagation();
+    })
+    )
+
+    mainSidebar.addEventListener('click',e=>{
+        e.stopPropagation(); // make sure the eventlistener on the body is not triggered when anything on the sidebar is clicked
+    });
+
+    document.body.addEventListener('click',(e)=>{
+        if(mainSidebar.className.includes('tr-drawer-modal')){
+            mainSidebar.classList.remove('active')
+            mainSidebar.parentElement.classList.remove('active');
+        }
+    })
+
+}
+catch(err){
+if(err.message.includes('Cannot read properties of null')) console.error(err);
+else throw err;
+}
+
+try{
+    //Toggle different variants of the component
+    const toggleVariantBtns = document.querySelectorAll('.toggle-variant-btn');
+    const tabToggleBtns = document.querySelectorAll('.main-demo-content .tabs-btn-list .tr-btn');
+    let currentActiveBlock = 1;
+
+        
+    let hideCurrentActiveBlocks = blockNumber => {
+        document.getElementById(`preview-block-${blockNumber}`).classList.add('hide');
+        document.getElementById(`code-block-${blockNumber}`).classList.add('hide');
+    }
+
+    let showBlocks = blockNumber =>{
+        document.getElementById(`preview-block-${blockNumber}`).classList.remove('hide');
+        document.getElementById(`code-block-${blockNumber}`).classList.remove('hide');
+    }
+
+    toggleVariantBtns.forEach(btn => {
+        btn.addEventListener('click',()=>{
+            let targetBlockNumber = btn.getAttribute('data-target');
+            hideCurrentActiveBlocks(currentActiveBlock);
+            showBlocks(targetBlockNumber);
+            currentActiveBlock = targetBlockNumber;
+
+            document.querySelector('.toggle-variant-btn.active').classList.remove('active'); //remove active from current active btn
+            btn.classList.add('active');
+        })
+    })    
+}
+catch(err){
+if(err.message.includes('Cannot read properties of null')) console.error(err);
+else throw err;
+}
```

Comment:
catch block Code can be indented for better readability

Reply:
Fixed in the latest commit